azureclient/credentialprovider/cli_flags.go

```diff
@@ -0,0 +1,133 @@
+package credentialprovider
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/services/resources/mgmt/2019-05-01/resources"
+	"github.com/Azure/go-autorest/autorest/azure/auth"
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/giantswarm/apiextensions/v3/pkg/apis/provider/v1alpha1"
+	"github.com/giantswarm/apiextensions/v3/pkg/label"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"sigs.k8s.io/cluster-api-provider-azure/api/v1alpha3"
+	ctrl "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type CLIFlagsCredentialProviderConfig struct {
+	CtrlClient ctrl.Client
+	Logger     micrologger.Logger
+
+	ManagementClusterClientID       string
+	ManagementClusterClientSecret   string
+	ManagementClusterSubscriptionID string
+	ManagementClusterTenantID       string
+}
+
+type CLIFlagsCredentialProvider struct {
```

Comment:
Is this implementation actually tied to CLI flags in any way?

Could this be `ManagementClusterCredentialProvider` ?

Reply:
renamed